src/client.js

```diff
@@ -224,6 +234,18 @@ exports.createClient = function(reference, name) {
         this._extData = new Buffer(JSON.stringify(ext)).toString('base64');
       }
     }
+
+    // If fake, we create an array this.fakeCalls[method] = [] for each method
+    if (this._options.fake) {
+      debug('Creating taskcluster-client object in "fake" mode');
+      this.fakeCalls = {};
+      reference.entries.filter(e => e.type === 'function').forEach(e => this.fakeCalls[e.name] = []);
+      // Warn if creating fakes in production, this just a warning to help people
+      // we can't know that everybody will set NODE_ENV to 'production'
+      if (process.env.NODE_ENV === 'production') {
+        console.log('Warning: taskcluster-client object created in "fake" mode');
```

Comment:
This should probably be an exception, too.

Reply:
Couldn't there be legitimate use-cases for this?

This was basically me being unsure, and therefor not willing force something on people.
If we throw an exception, then we have to make definition of certain environment variables are hard requirement.

As environment variables are global state, I think that's a bad choice.

-----
We always load all config values from `config.yml` which loads from environment variables. Thus, we can change things in testing. The only exception to this is `DEBUG` which is only used for debugging (really our services shouldn't be running with it). And possibly `NODE_ENV` which affects how `express` behaves, but we pretend we don't see that there is a difference between test and production there.